make-appimage.sh

```diff
@@ -209,13 +211,22 @@ linuxdeploy \
     --library=/usr/lib/$ARCH-linux-gnu/libthai.so.0 \
     --custom-apprun=$WORKSPACE/AppRun \
     --appdir $APPDIR \
-    --output appimage \
     --plugin gtk
 fi
 
 DATE_STR=$(date +%y%m%d%H%M)
 OUT_APPIMAGE="0ad-$VERSION-$DATE_STR-$ARCH.AppImage"
-mv 0_A.D.-$VERSION-$ARCH.AppImage $OUT_APPIMAGE
+
+REPO="0ad-appimage"
+TAG="latest"
+[ -z "$GITHUB_REPOSITORY_OWNER" ] && GITHUB_REPOSITORY_OWNER="0ad-matters"
+UPINFO="gh-releases-zsync|$GITHUB_REPOSITORY_OWNER|$REPO|$TAG|*$ARCH.AppImage.zsync"
+
+./appimagetool --appimage-extract-and-run \
```

Comment:
```suggestion
appimagetool --appimage-extract-and-run \
```

It's in the PATH, installed to $HOME/.local/bin https://github.com/andy5995/linuxdeploy-build-helper-container/blob/3f70f7b215753306b9474f3d95c777498a0c7186/Dockerfile#L99

Reply:
great

Reply:
Hmmm... failed. Strange that this option isn't available:

```
2025-02-01T01:46:41.2132353Z appimagetool --appimage-extract-and-run \
2025-02-01T01:46:41.2132703Z 	--comp zstd --mksquashfs-opt -Xcompression-level --mksquashfs-opt 20 \
2025-02-01T01:46:41.2133014Z 	-u "$UPINFO" \
2025-02-01T01:46:41.2133177Z 	"$APPDIR" "$OUT_APPIMAGE"
2025-02-01T01:46:41.2183371Z Option parsing failed: Unknown option --appimage-extract-and-run
2025-02-01T01:46:42.4148172Z ##[error]Process completed with exit code 1.
```

I'm installing it in the docker image  from https://github.com/AppImage/appimagetool

Could we just run that argument from the appimage first?

```
[andy@prometheus Downloads]$ ./0ad-0.27.0-2501310402-x86_64.AppImage --appimage-help
AppImage options:

  --appimage-extract [<pattern>]  Extract content from embedded filesystem image
                                  If pattern is passed, only extract matching
                                  files
  --appimage-extract-and-run      Temporarily extract content from embedded
                                  filesystem image, run contained application,
                                  then delete temporarily extracted content
  --appimage-help                 Print this help
  --appimage-mount                Mount embedded filesystem image and print
                                  mount point and wait for kill with Ctrl-C
  --appimage-offset               Print byte offset to start of embedded
                                  filesystem image
  --appimage-portable-home        Create a portable home folder to use as $HOME
  --appimage-portable-config      Create a portable config folder to use as
                                  $XDG_CONFIG_HOME
  --appimage-signature            Print digital signature embedded in AppImage
  --appimage-updateinfo[rmation]  Print update info embedded in AppImage
  --appimage-version              Print version of AppImage runtime
```

Reply:
that argument is used widelly in systems without a working FUSE, such as a docker container... I added that flag because before the command was a local file and worked. In this case, we should not need to extract and run the AppImage from $PATH